src/Bart/Optional/Absent.php

```diff
@@ -0,0 +1,100 @@
+<?php
+namespace Bart\Optional;
+
+use Bart\Exceptions\IllegalStateException;
+
+/**
+ * Optional package class that handles a nonexistent value. This
+ * class operates as a singleton.
+ * Class Absent
+ * @package Bart\Optional
+ */
+class Absent extends Optional
```

Comment:
Similar reasons as Present instead of Some. The other issue I felt with calling this None is that we're avoiding using null with this and None seemed a little too similar. If we're avoiding nulls, then it seemed less ideal to say a lack of a value was an instance of None than it was to say it was Absent.